subprojects/snapshots/src/test/groovy/org/gradle/internal/fingerprint/impl/FingerprintCompareStrategyTest.groovy

```diff
@@ -142,131 +123,183 @@ class FingerprintCompareStrategyTest extends Specification {
+        changes(strategy, includeAdded, [
+            "one": fingerprint("one"),
+            "two": fingerprint("two"),
+            "four": fingerprint("four")
+        ], [
+            "one": fingerprint("one"),
+            "three": fingerprint("three"),
+            "four": fingerprint("four")
+        ]) == results
 
         where:
         strategy | includeAdded | results
         ABSOLUTE | true         | [added("two"), removed("three")]
         ABSOLUTE | false        | [removed("three")]
     }
 
-    @Unroll
-    def "reordering (#strategy, include added: #includeAdded)"() {
+    def "reordering (NormalizedPathFingerprintCompareStrategy, include added: true)"() {
         expect:
-        changes(strategy, includeAdded,
-            ["one-new": fingerprint("one"), "two-new": fingerprint("two"), "three-new": fingerprint("three")],
-            ["one-old": fingerprint("one"), "three-old": fingerprint("three"), "two-old": fingerprint("two")]
-        ) == results
-
-        where:
-        strategy   | includeAdded | results
-        NORMALIZED | true         | []
-        NORMALIZED | false        | []
+        changes(NORMALIZED, true, [
+            "new/one": fingerprint("one"),
+            "new/two": fingerprint("two"),
+            "new/three": fingerprint("three")
+        ], [
+            "old/one": fingerprint("one"),
+            "old/three": fingerprint("three"),
+            "old/two": fingerprint("two")
+        ]) == []
     }
 
-    @Unroll
-    def "reordering with absolute paths (#strategy, include added: #includeAdded)"() {
+    def "reordering with absolute paths (#strategy.class.simpleName, include added: #includeAdded)"() {
         expect:
-        changes(strategy, includeAdded,
-            ["one": fingerprint("one"), "two": fingerprint("two"), "three": fingerprint("three")],
-            ["one": fingerprint("one"), "three": fingerprint("three"), "two": fingerprint("two")]
-        ) == results
+        changes(strategy, includeAdded, [
+            "one": fingerprint("one"),
+            "two": fingerprint("two"),
+            "three": fingerprint("three")
+        ], [
+            "one": fingerprint("one"),
+            "three": fingerprint("three"),
+            "two": fingerprint("two")
+        ]) == results
 
         where:
         strategy | includeAdded | results
         ABSOLUTE | true         | []
         ABSOLUTE | false        | []
     }
 
-    @Unroll
-    def "handling duplicates (#strategy, include added: #includeAdded)"() {
+    def "handling duplicates (NormalizedPathFingerprintCompareStrategy)"() {
         expect:
-        changes(strategy, includeAdded,
-            ["one-new-1": fingerprint("one"), "one-new-2": fingerprint("one"), "two-new": fingerprint("two")],
-            ["one-old-1": fingerprint("one"), "one-old-2": fingerprint("one"), "two-old": fingerprint("two")]
-        ) == []
+        changes(NORMALIZED, true, [
+            "new/one-1": fingerprint("one"),
```

Comment:
here the local and absolute file seems to be different, should I fix these as well?

Reply:
Please go ahead.